src/main.rs

```diff
@@ -8,11 +8,17 @@
 pub mod hardware;
 pub mod net;
 
+use defmt::{info, Format};
 use defmt_rtt as _; // global logger
 
-extern crate panic_halt;
+extern crate panic_probe;
```

Comment:
I think nowadays this is deprecated. https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#no-more-extern-crate

Recommend bumping `edition = "2021"` in Cargo.toml.